tests/php/Control/Middleware/RewriteHashLinksMiddlewareTest.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace SilverStripe\Control\Tests\Middleware;
+
+use SilverStripe\Control\Middleware\RewriteHashLinksMiddleware;
+use SilverStripe\Control\Director;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\Convert;
+use SilverStripe\Dev\SapphireTest;
+
+class RewriteHashLinksMiddlewareTest extends SapphireTest
+{
+    /**
+     * Tests rewriting of HTML content types to ensure that it's properly rewriting anchors when the base tag is present
+     */
+    public function testRewriteHTMLWithBase()
+    {
+        $_SERVER['HTTP_HOST'] = 'www.mysite.com';
+        $_SERVER['REQUEST_URI'] = '//file.com?foo"onclick="alert(\'xss\')""';
+
+        $base = Convert::raw2att('/file.com?foo"onclick="alert(\'xss\')""');
+
```

Comment:
Bunch of double blank lines in this file that should be reduces to a single blank line